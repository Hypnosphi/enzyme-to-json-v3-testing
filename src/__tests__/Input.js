import React from 'react';
import Enzyme, {shallow, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Input from '../Input';

Enzyme.configure({adapter: new Adapter()});

it('converts basic button using shallow', () => {
  const shallowed = shallow(<Input className="class" value="testVal" />);

  expect(shallowed).toMatchSnapshot();
});

it('converts basic button using mount', () => {
  const mounted = mount(<Input className="class" value="testVal" />);

  expect(mounted).toMatchSnapshot();
});

it('converts basic button using render', () => {
  const rendered = render(<Input className="class" value="testVal" />);

  expect(rendered).toMatchSnapshot();
});
