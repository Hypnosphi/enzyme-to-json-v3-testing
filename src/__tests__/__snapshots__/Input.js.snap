// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts basic button using mount 1`] = `
<Input
  className="class"
  value="testVal"
>
  <input
    className="class"
    value="testVal"
  />
</Input>
`;

exports[`converts basic button using render 1`] = `
<input
  class="class"
  value="testVal"
/>
`;

exports[`converts basic button using shallow 1`] = `
<input
  className="class"
  value="testVal"
/>
`;
